nx.json

```diff
@@ -299,28 +195,89 @@
         "{projectRoot}/dist/middlewares/**/*.d.ts",
         "{projectRoot}/dist/middlewares/**/*.d.ts.map",
         "{projectRoot}/build/.tsbuildinfo.builders"
-      ]
+      ],
+      "cache": true
     },
     "build-plugins": {
-      "dependsOn": [
-        "^build",
-        "compile"
-      ],
-      "inputs": [
-        "plugins"
-      ],
-      "outputs": [
-        "{projectRoot}/dist/plugins",
-        "{projectRoot}/build/plugins"
-      ]
+      "dependsOn": ["^build", "compile"],
+      "inputs": ["plugins"],
+      "outputs": ["{projectRoot}/dist/plugins", "{projectRoot}/build/plugins"],
+      "cache": true
     },
     "documentation": {
-      "outputs": [
-        "{workspaceRoot}/generated-doc/{projectName}/**/*"
-      ],
-      "inputs": [
-        "source"
-      ]
+      "outputs": ["{workspaceRoot}/generated-doc/{projectName}/**/*"],
+      "inputs": ["source"],
+      "cache": true
+    },
+    "prepare": {
+      "cache": true
+    },
+    "build-cjs": {
+      "cache": true
+    },
+    "build-esm2015": {
+      "cache": true
+    },
+    "build-esm2020": {
+      "cache": true
+    },
+    "build-react": {
+      "cache": true
+    },
+    "build-panel": {
+      "cache": true
+    },
+    "build-wrapper": {
+      "cache": true
+    },
+    "build-schematics": {
+      "cache": true
+    },
+    "build-extension": {
+      "cache": true
+    },
+    "build-components": {
+      "cache": true
+    },
+    "build-tools": {
+      "cache": true
+    },
+    "copy-assets": {
+      "cache": true
+    },
+    "generate-translations": {
+      "cache": true
+    },
+    "extract-translations": {
+      "cache": true
+    },
+    "extract-components": {
+      "cache": true
+    },
+    "extract-style": {
+      "cache": true
+    },
+    "extract-rules-engine": {
+      "cache": true
+    },
+    "build-swagger": {
+      "cache": true
+    },
+    "i18n": {
+      "cache": true
+    },
+    "@nx/jest:jest": {
```

Comment:
This is the name of a `executor`, not a `task. Do rules can be applied according to executor now? Should them be defined in `targetDefaults` as well?

Reply:
yes executor can be used in targetDefaults according to the docs but it seems that Nx won't try to read the config for the task name if a config for the executor is found so I think I should remove that part

https://nx.dev/reference/nx-json#target-defaults
> Note that the inputs and outputs are respecified on the @nx/js:tsc default configuration. This is required, as when reading target defaults Nx will only ever look at one key. If there is a default configuration based on the executor used, it will be read first. If not, Nx will fall back to looking at the configuration based on target name. For instance, running nx build project will read the options from targetDefaults[@nx/js:tsc] if the target configuration for build uses the @nx/js:tsc executor. It would not read any of the configuration from the build target default configuration unless the executor does not match.
```